inc/tasks/userpruning.php

```diff
@@ -41,11 +41,25 @@ function task_userpruning($task)
 			unset($in_usergroups[$key]);
 		}
 
+		$prunepostcount = (int)$mybb->settings['prunepostcount'];
+
 		$regdate = TIME_NOW-((int)$mybb->settings['dayspruneregistered']*24*60*60);
-		$query = $db->simple_select("users", "uid", "regdate <= ".(int)$regdate." AND postnum <= ".(int)$mybb->settings['prunepostcount']." AND usergroup IN(".$db->escape_string(implode(',', $in_usergroups)).")");
-		while($user = $db->fetch_array($query))
+		$query = $db->simple_select('users', 'uid', "regdate<={$regdate} AND postnum<={$prunepostcount} AND usergroup IN({$db->escape_string(implode(',', $in_usergroups))})");
+		while($uid = (int)$db->fetch_field($query, 'uid'))
```

Comment:
Not an issue (except increasing the file size and execution time a very tiny bit) but still doesn't make much sense to use it here. It's not an user input so the extra security layer is never needed in this case, unless someone hacks the database, changes the `uid` column to another type, injects malicious strings and then runs this query, which would be the weirdest attack ever.